src/main/scala/com/github/xiaodongw/swagger/finatra/FinatraSwagger.scala

```diff
@@ -14,24 +26,199 @@ object FinatraSwagger {
   implicit def convertToFinatraSwagger(swagger: Swagger): FinatraSwagger = new FinatraSwagger(swagger)
 }
 
+sealed trait ModelParam {
+  val name: String
+  val description: String
+  val required: Boolean
+  val typ: Class[_]
+}
+
+sealed trait FinatraParams
+case class Route(name: String, typ: Class[_], description: String = "", required: Boolean = true) extends FinatraParams with ModelParam
+case class Query(name: String, typ: Class[_], description: String = "", required: Boolean = true) extends FinatraParams with ModelParam
+case class Body(description: String = "", name: String, typ: Class[_]) extends FinatraParams
+case class RequestInject(name: String) extends FinatraParams
+case class Header(name: String, required: Boolean = true, description: String = "", typ: Class[_]) extends FinatraParams with ModelParam
+case class Form(name: String, description: String = "", required: Boolean = true, typ: Class[_]) extends FinatraParams with ModelParam
+
+object Resolvers {
+  class ScalaOptionResolver(objectMapper: ObjectMapper) extends ModelResolver(objectMapper) {
```

Comment:
This seems not working, the model generate from your example is wrong.

I saw the Option support actually has been fixed in swagger-scala-module (https://github.com/swagger-api/swagger-scala-module/blob/ce25f52a8839592137e414bad06db87d4332d670/src/main/scala/io/swagger/scala/converter/SwaggerScalaModelConverter.scala), we need it to publish a new release.